src/views/AddTimer.tsx

```diff
@@ -0,0 +1,183 @@
+// AddTimer.tsx
+
+import { faPlus, faTimes } from '@fortawesome/free-solid-svg-icons';
+import { useState } from 'react';
+import type React from 'react';
+import { useNavigate } from 'react-router-dom';
+import { v4 as uuidv4 } from 'uuid';
+import Button from '../components/button/Button';
+import { useTimerContext } from '../context/TimerContext';
+
+export type TimerType = 'stopwatch' | 'countdown' | 'xy' | 'tabata';
+
+const AddTimer = () => {
+    const [timerType, setTimerType] = useState<TimerType>('stopwatch');
+    const [minutes, setMinutes] = useState(0);
```

Comment:
A lot of state hooks here. You could consider collapsing them a little bit (ie. `const [roundTime, setRoundTime] = useState({minutes: 1, seconds: 0})`)
or if you break out the renderTimeInput function into a component (see comment below), you may be able to bring some of these state hooks down into that component.